tests/jballerina-unit-test/src/test/resources/test-src/isolation-analysis/isolation_analysis.bal

```diff
@@ -543,6 +543,54 @@ function testIsolatedFunctionWithSelfAsCapturedVariable() {
     assertEquality(false, b.compare());
 }
 
+isolated function testIsolatedFPCallInIsolatedFunction() {
```

Comment:
Something not directly related to the change
```ballerina
isolated function dummyTest() {
    var f = abc;
    // f(); // @error invalid invocation of a non-isolated function in an 'isolated' function
    print(<any>f is isolated function); //true
}

function abc() {
}
```

Is it a conscious decision to throw the error even though inferring would resolve it as isolated?

Reply:
Yes. 

https://ballerina.io/spec/lang/master/#section_8.3.3

> This inference is purely an optimization to improve service concurrency and is subject to two constraints. First, any inferences do not affect the static types visible to other modules. Second, **anything explicitly declared as isolated must satisfy the requirements of this specification without without relying on inference.**